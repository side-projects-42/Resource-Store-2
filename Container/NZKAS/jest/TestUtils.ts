/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

// eslint-disable-next-line import/no-extraneous-dependencies
import {Config} from '@jest/types';

const DEFAULT_GLOBAL_CONFIG: Config.GlobalConfig = {
  bail: 0,
  changedFilesWithAncestor: false,
  changedSince: '',
  collectCoverage: false,
  collectCoverageFrom: [],
  collectCoverageOnlyFrom: null,
  coverageDirectory: 'coverage',
  coverageReporters: [],
  coverageThreshold: {global: {}},
  detectLeaks: false,
  detectOpenHandles: false,
  enabledTestsMap: null,
  errorOnDeprecated: false,
  expand: false,
  extraGlobals: [],
  filter: null,
  findRelatedTests: false,
  forceExit: false,
  globalSetup: null,
  globalTeardown: null,
  json: false,
  lastCommit: false,
  listTests: false,
  logHeapUsage: false,
  maxConcurrency: 5,
  maxWorkers: 2,
  noSCM: null,
  noStackTrace: false,
  nonFlagArgs: [],
  notify: false,
  notifyMode: 'failure-change',
  onlyChanged: false,
  onlyFailures: false,
  outputFile: null,
  passWithNoTests: false,
  projects: [],
  replname: null,
  reporters: [],
  rootDir: '/test_root_dir/',
  runTestsByPath: false,
  silent: false,
  skipFilter: false,
  testFailureExitCode: 1,
  testNamePattern: '',
  testPathPattern: '',
  testResultsProcessor: null,
  updateSnapshot: 'none',
  useStderr: false,
  verbose: false,
  watch: false,
  watchAll: false,
  watchPlugins: [],
  watchman: false,
};

const DEFAULT_PROJECT_CONFIG: Config.ProjectConfig = {
  automock: false,
  browser: false,
  cache: false,
  cacheDirectory: '/test_cache_dir/',
  clearMocks: false,
  coveragePathIgnorePatterns: [],
  cwd: '/test_root_dir/',
  detectLeaks: false,
  detectOpenHandles: false,
  displayName: undefined,
  errorOnDeprecated: false,
  extraGlobals: [],
  filter: null,
  forceCoverageMatch: [],
  globalSetup: null,
  globalTeardown: null,
  globals: {},
  haste: {
    providesModuleNodeModules: [],
  },
  moduleDirectories: [],
  moduleFileExtensions: ['js'],
  moduleLoader: '/test_module_loader_path',
  moduleNameMapper: [],
  modulePathIgnorePatterns: [],
  modulePaths: [],
  name: 'test_name',
  prettierPath: 'prettier',
  resetMocks: false,
  resetModules: false,
  resolver: null,
  restoreMocks: false,
  rootDir: '/test_root_dir/',
  roots: [],
  runner: 'jest-runner',
  setupFiles: [],
  setupFilesAfterEnv: [],
  skipFilter: false,
  skipNodeResolution: false,
  snapshotResolver: null,
  snapshotSerializers: [],
  testEnvironment: 'node',
  testEnvironmentOptions: {},
  testLocationInResults: false,
  testMatch: [],
  testPathIgnorePatterns: [],
  testRegex: ['\\.test\\.js$'],
  testRunner: 'jest-jasmine2',
  testURL: '',
  timers: 'real',
  transform: [],
  transformIgnorePatterns: [],
  unmockedModulePathPatterns: null,
  watchPathIgnorePatterns: [],
};

export const makeGlobalConfig = (
  overrides: Partial<Config.GlobalConfig> = {},
): Config.GlobalConfig => {
  const overridesKeys = new Set(Object.keys(overrides));
  Object.keys(DEFAULT_GLOBAL_CONFIG).forEach(key => overridesKeys.delete(key));

  if (overridesKeys.size > 0) {
    throw new Error(`
      Properties that are not part of GlobalConfig type were passed:
      ${JSON.stringify(Array.from(overridesKeys))}
    `);
  }

  return {...DEFAULT_GLOBAL_CONFIG, ...overrides};
};

export const makeProjectConfig = (
  overrides: Partial<Config.ProjectConfig> = {},
): Config.ProjectConfig => {
  const overridesKeys = new Set(Object.keys(overrides));
  Object.keys(DEFAULT_PROJECT_CONFIG).forEach(key => overridesKeys.delete(key));

  if (overridesKeys.size > 0) {
    throw new Error(`
      Properties that are not part of ProjectConfig type were passed:
      ${JSON.stringify(Array.from(overridesKeys))}
    `);
  }

  return {...DEFAULT_PROJECT_CONFIG, ...overrides};
};
