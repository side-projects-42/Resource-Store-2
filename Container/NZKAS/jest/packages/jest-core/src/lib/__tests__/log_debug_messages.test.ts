/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import {wrap} from 'jest-snapshot-serializer-raw';
import logDebugMessages from '../log_debug_messages';
import {makeGlobalConfig, makeProjectConfig} from '../../../../../TestUtils';

jest.mock('../../../package.json', () => ({version: 123}));

jest.mock('myRunner', () => ({name: 'My Runner'}), {virtual: true});

const getOutputStream = (resolve: (message: string) => void) =>
  ({
    write(message: string) {
      resolve(message);
    },
  } as NodeJS.WriteStream);

it('prints the jest version', async () => {
  expect.assertions(1);
  const message = await new Promise<string>(resolve => {
    logDebugMessages(
      makeGlobalConfig({watch: true}),
      makeProjectConfig({testRunner: 'myRunner'}),
      getOutputStream(resolve),
    );
  });

  expect(JSON.parse(message).version).toBe(123);
});

it('prints the test framework name', async () => {
  expect.assertions(1);
  const message = await new Promise<string>(resolve => {
    logDebugMessages(
      makeGlobalConfig({watch: true}),
      makeProjectConfig({testRunner: 'myRunner'}),
      getOutputStream(resolve),
    );
  });

  expect(JSON.parse(message).configs.testRunner).toBe('myRunner');
});

it('prints the config object', async () => {
  expect.assertions(1);
  const globalConfig = makeGlobalConfig({
    watch: true,
  });
  const config = makeProjectConfig({
    automock: false,
    rootDir: '/path/to/dir',
    roots: ['path/to/dir/test'],
    testRunner: 'myRunner',
  });
  const message = await new Promise<string>(resolve => {
    logDebugMessages(globalConfig, config, getOutputStream(resolve));
  });
  expect(wrap(message)).toMatchSnapshot();
});
