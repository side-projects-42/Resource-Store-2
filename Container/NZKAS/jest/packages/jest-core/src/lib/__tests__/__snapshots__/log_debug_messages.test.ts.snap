// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prints the config object 1`] = `
{
  "configs": {
    "automock": false,
    "browser": false,
    "cache": false,
    "cacheDirectory": "/test_cache_dir/",
    "clearMocks": false,
    "coveragePathIgnorePatterns": [],
    "cwd": "/test_root_dir/",
    "detectLeaks": false,
    "detectOpenHandles": false,
    "errorOnDeprecated": false,
    "extraGlobals": [],
    "filter": null,
    "forceCoverageMatch": [],
    "globalSetup": null,
    "globalTeardown": null,
    "globals": {},
    "haste": {
      "providesModuleNodeModules": []
    },
    "moduleDirectories": [],
    "moduleFileExtensions": [
      "js"
    ],
    "moduleLoader": "/test_module_loader_path",
    "moduleNameMapper": [],
    "modulePathIgnorePatterns": [],
    "modulePaths": [],
    "name": "test_name",
    "prettierPath": "prettier",
    "resetMocks": false,
    "resetModules": false,
    "resolver": null,
    "restoreMocks": false,
    "rootDir": "/path/to/dir",
    "roots": [
      "path/to/dir/test"
    ],
    "runner": "jest-runner",
    "setupFiles": [],
    "setupFilesAfterEnv": [],
    "skipFilter": false,
    "skipNodeResolution": false,
    "snapshotResolver": null,
    "snapshotSerializers": [],
    "testEnvironment": "node",
    "testEnvironmentOptions": {},
    "testLocationInResults": false,
    "testMatch": [],
    "testPathIgnorePatterns": [],
    "testRegex": [
      "\\\\.test\\\\.js$"
    ],
    "testRunner": "myRunner",
    "testURL": "",
    "timers": "real",
    "transform": [],
    "transformIgnorePatterns": [],
    "unmockedModulePathPatterns": null,
    "watchPathIgnorePatterns": []
  },
  "globalConfig": {
    "bail": 0,
    "changedFilesWithAncestor": false,
    "changedSince": "",
    "collectCoverage": false,
    "collectCoverageFrom": [],
    "collectCoverageOnlyFrom": null,
    "coverageDirectory": "coverage",
    "coverageReporters": [],
    "coverageThreshold": {
      "global": {}
    },
    "detectLeaks": false,
    "detectOpenHandles": false,
    "enabledTestsMap": null,
    "errorOnDeprecated": false,
    "expand": false,
    "extraGlobals": [],
    "filter": null,
    "findRelatedTests": false,
    "forceExit": false,
    "globalSetup": null,
    "globalTeardown": null,
    "json": false,
    "lastCommit": false,
    "listTests": false,
    "logHeapUsage": false,
    "maxConcurrency": 5,
    "maxWorkers": 2,
    "noSCM": null,
    "noStackTrace": false,
    "nonFlagArgs": [],
    "notify": false,
    "notifyMode": "failure-change",
    "onlyChanged": false,
    "onlyFailures": false,
    "outputFile": null,
    "passWithNoTests": false,
    "projects": [],
    "replname": null,
    "reporters": [],
    "rootDir": "/test_root_dir/",
    "runTestsByPath": false,
    "silent": false,
    "skipFilter": false,
    "testFailureExitCode": 1,
    "testNamePattern": "",
    "testPathPattern": "",
    "testResultsProcessor": null,
    "updateSnapshot": "none",
    "useStderr": false,
    "verbose": false,
    "watch": true,
    "watchAll": false,
    "watchPlugins": [],
    "watchman": false
  },
  "version": 123
}

`;
